### PR TITLE
Make composer installs on travis more efficient

### PR DESCRIPTION
**Pull request by dhensby**
Use our more standard travis layout

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,20 +22,27 @@ matrix:
       env: DB=MYSQL PDO=1 PHPUNIT_COVERAGE_TEST=1
 
 before_script:
-  - composer validate
-  - composer self-update || true
-  - "if [ \"$PHPCS_TEST\" = \"1\" ]; then pyrus install pear/PHP_CodeSniffer; fi"
+# Init PHP
+  - export CORE_RELEASE=$TRAVIS_BRANCH
```
Comment (tractorcow) [inline .travis.yml]: We don't need to keep adding in CORE_RELASE env anymore. Shall we drop this in the future? It was only necessary for travis-support-module.

File: .travis.yml [hunk 2]
```diff
@@ -22,20 +22,27 @@ matrix:
...
+  - if [[ $PHPUNIT_TEST ]]; then vendor/bin/phpunit tests; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then phpdbg -qrr vendor/bin/phpunit --coverage-clover=coverage.xml tests; fi
+  - if [[ $PHPCS_TEST ]]; then composer run-script lint; fi
 
 after_success:
-  - "if [ \"$PHPUNIT_COVERAGE_TEST\" = \"1\" ]; then bash <(curl -s https://codecov.io/bash) -f coverage.xml; fi"
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then bash <(curl -s https://codecov.io/bash) -f coverage.xml; fi
+
+after_failure:
+  - if [[ $BEHAT_TEST ]]; then php ./framework/tests/behat/travis-upload-artifacts.php --if-env BEHAT_TEST,ARTIFACTS_BUCKET,ARTIFACTS_KEY,ARTIFACTS_SECRET --target-path $TRAVIS_REPO_SLUG/$TRAVIS_BUILD_ID/$TRAVIS_JOB_ID --artifacts-base-url https://s3.amazonaws.com/$ARTIFACTS_BUCKET/ --artifacts-path ./artifacts/; fi
```
Comment (tractorcow) [inline .travis.yml]: We don't have behat tests for this module.

File: .travis.yml (hunk 1 above)
Comment (tractorcow) [inline .travis.yml]: We don't need to keep adding in CORE_RELASE env anymore. Shall we drop this in the future? It was only necessary for travis-support-module.

**Comment by dhensby** [comment]
feedback addresed

**Pull request merged**